### PR TITLE
BREAKING: Delete NavigationList, introduce LinkList component

### PR DESCRIPTION
**Pull request by mainframev**
<br/><br/><br/><url>LiveURL: https://orbit-components-mainframe-navigation-components.surge.sh</url>

### CODE REVIEW

**Comment by mainframev** [comment]
@williamkolmacka as soon as staging is ready, could you please check if it's ok with design? :) 🙏

File: src/List/README.md [hunk 1]
```diff
@@ -48,3 +48,28 @@ Table below contains all types of the props in ListItem component.
 #### Functional specs
 
 - You can color your icon if you pass some value into `color` or `customColor` prop of the [Icon](./Icon). Be aware of using other components, because they are not styled by default.
+
+<LinkList type="" spacing="loose">
```
Comment (vepor) [inline src/List/README.md]: Please create new MD file for the LinkList 🙏

File: src/Modal/Modal.stories.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ storiesOf("Modal", module)
         <Modal onClose={onClose} size={size}>
           <ModalHeader title={title}>{description}</ModalHeader>
           <ModalSection>
-            <Text>{content}</Text>
+            <ButtonLink transparent>asdaddas</ButtonLink>
```
Comment (vepor) [inline src/Modal/Modal.stories.js]: Revert 🙏

File: src/index.js [hunk 3]
```diff
@@ -44,8 +44,6 @@ export { default as Slider } from "./Slider";
 
 // Navigation components
 export { default as NavigationBar } from "./NavigationBar";
-export { default as NavigationList } from "./NavigationList";
-export { default as NavigationListItem } from "./NavigationList/NavigationListItem";
 export { default as NavigationDrawer } from "./Drawer";
```
Comment (vepor) [inline src/index.js]: Add export of `LinkList` 🙏

File: src/List/README.md (hunk 1 above)
Comment (vepor) [reply src/List/README.md]: Not resolved, cc @mainframev

**Pull request merged**